src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2324,11 +2321,6 @@ PeerImp::addLedger(
         recentLedgers_.end())
         return;
 
-    // VFALCO TODO See if a sorted vector would be better.
```

Comment:
I'm wondering why this comment is here. It seems that circular_buffer is the obvious choice here, so I'm wondering why Vinny was suggesting a sorted vector.

Reply:
We use `std::find` in a few places, so a sorted vector would make it possible to use [`std::binary_search`](https://en.cppreference.com/w/cpp/algorithm/binary_search) instead of `std::find`.

It would trade additional insertion complexity for fewer comparisons during a search. I don't know how this comes down.

Reply:
I've done a few studies on this and it depends on the length of the sequence, complexity of the comparison operator and size of each element.  But the takeaway is that if sizeof and comparison are fairly cheap, the linear search wins for a surprising length.  I don't have the exact numbers off the top of my head, but I believe it was in the hundreds, not in the tens, before binary search began to win.